src/Service/CacheService.php

```diff
@@ -875,8 +875,13 @@ public function retrieveObjectsFromCache(array $filter, array $options, array $c
         $this->session->set('mongoDBFilter', $filter);
 
         $collection = $this->client->objects->json;
-        $results    = $collection->find($filter, $options)->toArray();
         $total      = $collection->count($filter);
+        
+        // If options is null, we only count the objects.
+        if ($options === null) {
+            return $total;
+        }
+        $results    = $collection->find($filter, $options)->toArray();
```

Comment:
I'd create a new function to be able to return the counts and split away fetching the results.

Reply:
But the function will be the same for 99%